### PR TITLE
[Fabric] Add keyboard handlers to ScrollView

### LINKED ISSUE #12076: Fabric ScrollView does not respond to keyboard commands (page up, page down, home, end, or arrow keys)
Related to https://github.com/microsoft/react-native-windows/issues/11152
Related to https://github.com/microsoft/react-native-windows/issues/12075

### PR DESCRIPTION
**Pull request by acoates-ms**
## Description
ScrollView does not respond to keyboard commands (page up, page down, home, end, or arrow keys).
Also fixes an issue where the scrollTo command would not scale the target location for scale factor, and ensures that scrollwheel handling chains through multiple scrollviews correctly.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
Resolves #12076 and #12074

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12190)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp [hunk 1]
```diff
@@ -272,37 +274,163 @@ void ScrollViewComponentView::OnPointerDown(const winrt::Windows::UI::Input::Poi
...
+}
+
+bool ScrollViewComponentView::pageUp(bool animate) noexcept {
+  if (m_scrollVisual.ScrollPosition().y <= 0.0f) {
+    return false;
+  }
+
+  float scrollToVertical =
+      m_scrollVisual.ScrollPosition().y - (m_layoutMetrics.frame.size.height * m_layoutMetrics.pointScaleFactor);
+  float scrollToHorizontal = m_scrollVisual.ScrollPosition().x;
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp:348]: Do these need to be min-ed so they don't scroll beyond 0?

Comment (chrisglein) [reply vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp:348]: Should this use `scrollDown` like `lineUp` does?

File: vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp [hunk 2]
```diff
@@ -272,37 +274,163 @@ void ScrollViewComponentView::OnPointerDown(const winrt::Windows::UI::Input::Poi
...
+  m_scrollVisual.TryUpdatePosition({scrollToHorizontal, scrollToVertical, 0.0f}, false /*animate*/);
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp:349]: Why isn't `animate` passed through?
Plenty more below. Same feedback for all.

File: vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp [hunk 3]
```diff
@@ -272,37 +274,163 @@ void ScrollViewComponentView::OnPointerDown(const winrt::Windows::UI::Input::Poi
...
+  }
+
+  float scrollToVertical = m_scrollVisual.ScrollPosition().y;
+  float scrollToHorizontal =
+      m_scrollVisual.ScrollPosition().x - (c_scrollerLineDelta * m_layoutMetrics.pointScaleFactor);
+  m_scrollVisual.TryUpdatePosition({scrollToHorizontal, scrollToVertical, 0.0f}, false /*animate*/);
+  return true;
+}
+
+bool ScrollViewComponentView::lineRight(bool animate) noexcept {
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp:386]: If the scroll view isn't set to be horizontal is this set up to correctly do nothing?

**Review by chrisglein — APPROVED** [review]

File: vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp (hunk 2 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp]: reworked the code to animate keyboarding - which seems to align with XAML's default behavior.
I also reworked the animate scroll behavior to alter the duration of the animation to be based on the distance to travel, also aligning more with XAML's behavior.

File: vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp (hunk 3 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Fabric/Composition/ScrollViewComponentView.cpp:350]: Right now it'll scroll horizontally if the content is larger than the viewport.  -- There are a bunch of properties to control scroll enable/disable which are not yet implemented.

**Review by jonthysell — APPROVED** [review]

**Pull request merged**